### PR TITLE
Add emoji search

### LINKED ISSUE #35: Emoji search
Would be quite useful if there was a way to search for emojis.

Switch to Emoji view and then search (e.g. type "heart" and you will see the heart emoji ❤️).

### PR DESCRIPTION
**Pull request by odmfl**
Added emoji search, from kitten board. Currently it works with English words, because there is a .json file in English only, other .json files should be added for other languages ​​as well it should work. It is based on this library
https://github.com/muan/emojilib

I tried and it works, take a look if you need to make any changes go ahead. I adapted the commits from kittenboard. I think it's a good starting point then others will be able to enhance this function

Fix #35 #197 #527

### CODE REVIEW

**Comment by ghost** [comment]
Why hasn't this been merged yet?

**Comment by logical-view** [comment]
> Hi, i just discover Openboard, the best keyboard i've ever tried on Android, and open-source, great job !

> The only big problem is the suggestions for me, especially on Vivaldi browser : in the URL bar, no suggestions, but right now, while i'm writing here, suggestions work. (As someone mentioned, it doesn't work in Google Play either, but i don't care, it's mainly disabled in the browser)
> 
> When do you think it will be fixed ? Next version ?
> Thanks!

()

When? Commit _739a8dde37d56f961ffdbafa707d674998d8e340_ ?

**Comment by tkterris** [comment]
Would it make sense to exclude emojis from auto correct? When I type "Ill" (as the first word in e.g. "I'll see you tomorrow") it gets auto corrected to the emoji "😷". Similarly if I enter "english" (corrected to "🏴󠁧󠁢󠁥󠁮󠁧󠁿") and a couple others.

File: app/src/main/java/org/dslul/openboard/inputmethod/latin/Suggest.java [hunk 1]
```diff
@@ -263,6 +266,20 @@ private void getSuggestedWordsForNonBatchInput(final WordComposer wordComposer,
         if (!TextUtils.isEmpty(typedWordString)) {
             suggestionsContainer.add(0, typedWordInfo);
         }
+		
+		if (!TextUtils.isEmpty(typedWordString)) {
+            List<String> results = EmojiSearch.getInstance().searchExact(typedWordString);
+            if (!results.isEmpty()) {
+                for (int idx = 0; idx < Math.min(2, results.size()); idx++) {
+                    suggestionsContainer.add(idx + 1, new SuggestedWordInfo(results.get(idx),
```
Comment (tkterris) [inline app/src/main/java/org/dslul/openboard/inputmethod/latin/Suggest.java:273]: A hacky solution to the issue mentioned in my comment is to insert emojis after the auto-correct entry. So:
```
suggestionsContainer.add(SuggestedWords.INDEX_OF_AUTO_CORRECTION + idx + 1, new SuggestedWordInfo(results.get(idx),
```

Comment (logical-view) [reply app/src/main/java/org/dslul/openboard/inputmethod/latin/Suggest.java:274]: Add less than three (**1-2**) emoji inside suggestions line, please!

(...correction word) Emoji, correction word, emoji (correction word...) **?**
--->  ` | ` :arrow_lower_left: ` | ` Arrow ` | ` :arrow_lower_right: ` | `

**Comment by unique-EJ** [comment]
Add less than three (**1-2**) emoji inside suggestions line, please!

(...correction word) Emoji, correction word, emoji (correction word...) **?**
--->  ` | ` :arrow_lower_left: ` | ` Arrow ` | ` :arrow_lower_right: ` | `

**Comment by RHJihan** [comment]
@odmfl, could you please clone your pull requests here? [https://github.com/Helium314/openboard/](https://github.com/Helium314/openboard/)

**Comment by odmfl** [comment]
Guys, I'm very busy, however I created a PR in the fork of @Helium314 
https://github.com/Helium314/openboard
hoping that he manages to continue the project with his fork.
Gesture typing should also be added which works fine here without external closed source libraries, the code could also be revised 
https://github.com/wordmage/openboard

**Comment by fehnomenal** [comment]
I'm using the fork right now and gesture typing as well as multi lang typing works like a charm 👍

**Comment by RHJihan** [comment]
@odmfl, I guess you should close this here. All your commits in the master branch are causing a notification to the participants.

**Comment by unique-EJ** [comment]
@odmfl rename your fork package to openboard-renewed / openboard-updated? _(in to-do)_
> A renewed OpenBoard keyboard, FOSS, updated to integrate changes / improvements that have been [...]

**Pull request closed without merging**